lib/orderbook/OrderBook.ts

```diff
@@ -434,6 +442,12 @@ class OrderBook extends EventEmitter {
         this.pool.broadcastOrderInvalidation(removeResult.order, takerPubKey);
       }
 
+      if (takerPubKey && !removeResult.fullyRemoved) {
```

Comment:
See comment above.
the condition on takerPubKey looks not in place. 
Also, why not to clear to hold amount even if the order is fully removed? Yes, it is redundant but the code will be much clearer.

I also  see no reason why not to hold/release for local matched.